pkg/resourcemanager/controller/tokenrequestor/reconciler.go

```diff
@@ -142,20 +142,50 @@ func (r *reconciler) reconcileServiceAccount(ctx context.Context, secret *corev1
 }
 
 func (r *reconciler) reconcileSecret(ctx context.Context, secret *corev1.Secret, token string, renewDuration time.Duration) error {
-	patch := client.MergeFrom(secret.DeepCopy())
+	if targetSecret := getTargetSecretFromAnnotations(secret.Annotations); targetSecret != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.targetClient, targetSecret, r.populateToken(targetSecret, token, renewDuration)); err != nil {
+			return err
+		}
 
-	if secret.Data == nil {
-		secret.Data = make(map[string][]byte, 1)
+		patch := client.MergeFrom(secret.DeepCopy())
+		if err := r.depopulateToken(secret)(); err != nil {
+			return err
+		}
+		return r.client.Patch(ctx, secret, patch)
 	}
 
-	if err := updateTokenInSecretData(secret.Data, token); err != nil {
+	patch := client.MergeFrom(secret.DeepCopy())
+	if err := r.populateToken(secret, token, renewDuration)(); err != nil {
```

Comment:
I assume this is required in case one switches from source -> target secret. How about the other way around? In addition, I think it makes sense to delete the secret in the target cluster when the source secret is deleted.

Reply:
> How about the other way around? 
> I think it makes sense to delete the secret in the target cluster when the source secret is deleted.

That's both true, however, for complexity reasons I would vote for not pursuing these parts now. We have already consciously decided not to delete the `ServiceAccount` in the target cluster (see https://github.com/gardener/gardener/pull/4931/commits/4c1178d91f06063967276575fc729d542212b3ea), so from that perspective it's not getting worse.
I'm still not happy with it myself, however, given that the proper solution is much more complicated (also for the "switch from target -> source secret scenario") while there is not a single use-case for it, I would not do it now and rather focus on the more valuable improvements (like #5121). OK?

Reply:
Okay, thanks for explaining.